File: 缓存UIWebView/oc/NSURLProtocolOC-webView/NSURLProtocolOC-webView/MHCacheTool/MHCacheModel.m
//
//  MHCacheModel.m
//  NSURLProtocolOC-webView
//
//  Created by 胡明昊 on 16/12/26.
//  Copyright © 2016年 CMCC. All rights reserved.
//

#import "MHCacheModel.h"

@implementation MHCacheModel

- (void)encodeWithCoder:(NSCoder *)aCoder {
    
    [aCoder encodeObject:self.data forKey:@"data"];
    [aCoder encodeObject:self.response forKey:@"response"];
    [aCoder encodeObject:self.redirectRequest forKey:@"redirectRequest"];
}


- (instancetype)initWithCoder:(NSCoder *)coder
{
    self = [super init];
    if (self) {
        _data = [coder decodeObjectForKey:@"data"];
        _response = [coder decodeObjectForKey:@"response"];
        _redirectRequest = [coder decodeObjectForKey:@"redirectRequest"];
    }
    return self;
}

@end

File: 缓存UIWebView/swift/NSURLProtocol-webView/NSURLProtocol-webView/LoadCacheTool/MHCacheModel.swift
//
//  MHCacheModel.swift
//  NSURLProtocol-webView
//
//  Created by 胡明昊 on 16/12/26.
//  Copyright © 2016年 CMCC. All rights reserved.
//

import UIKit

class MHCacheModel: NSObject,NSCoding {
    
    var data: Data?
    var response: URLResponse?
    var redirectRequest: URLRequest?
    
    /// 归档
    func encode(with aCoder: NSCoder) {
        aCoder.encode(data, forKey: "data")
        aCoder.encode(response, forKey: "response")
        aCoder.encode(redirectRequest, forKey: "redirectRequest")
    }
    
    
    /// 解档
    required init?(coder aDecoder: NSCoder) {
        super.init()
        
        data = aDecoder.decodeObject(forKey: "data") as! Data?
        response = aDecoder.decodeObject(forKey: "response") as! URLResponse?
        redirectRequest = aDecoder.decodeObject(forKey: "redirectRequest") as! URLRequest?
    }
    
    
    override init() {
        super.init()
    }
}
